### PR TITLE
session: skip global variable cache in unit test

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Make the CI faster !

```
➜  session git:(accelerate-ci-sleep) ✗ GO111MODULE=on go test -check.f TestSetGlobalTZ
before:
ok  	github.com/pingcap/tidb/session	3.131s
after:
ok  	github.com/pingcap/tidb/session	0.109s
```

### What is changed and how it works?

If the global variable cache is used, the test code has to sleep a while
to ensure it load the fresh data. This may slow the CI

### Check List 

Tests 

 - No code

PTAL @jackysp @winkyao

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8594)

### CODE REVIEW

**Comment by lysu** [comment]
@winoros it also help `TestPrepareOverMaxPreparedStmtCount`

File: session/tidb_test.go [hunk 1]
```diff
@@ -34,6 +34,10 @@ import (
 	"github.com/pingcap/tidb/util/testleak"
 )
 
+func init() {
+	runningUnitTest = true
```
Comment (eurekaka) [inline session/tidb_test.go]: `TestSQLMode` in executor package wants to set this variable also, can we export it here?

Comment (zz-jason) [reply session/tidb_test.go]: many packages need this variable to accelerate CI 😂

Comment (tiancaiamao) [reply session/tidb_test.go]: Done

**Pull request merged**